autobot/github.py

```diff
@@ -17,232 +17,217 @@
+    """Pull request wrapper."""
+
+    def __init__(self, pr, maintainers=None):
+        """Pull request wrapper object initialization."""
+        self._gh_obj = pr
+        self.maintainers = maintainers
+
+    @property
+    def info(self):
+        """Fetch information for the pull request."""
         return {
-            "id": pr.number,
-            "url": pr.html_url,
-            "title": pr.title,
-            "creation_date": pr.created_at,
-            "description": pr.body,
-            "state": pr.state,
-            # 'labels': [
-            #    {
-            #          'name': label.name,
-            #          'color': label.color,
-            #          'url': label.url
-            #    } for label in pr.labels()
-            #    ],
-            "user": {"name": pr.user.login, "url": pr.user.url},
-            "related_issue": pr.issue_url,
+            "id": self.number,
+            "url": self.html_url,
+            "title": self.title,
+            "creation_date": self.created_at,
+            "description": self.body,
+            "state": self.state,
+            "user": {"name": self.user.login, "url": self.user.html_url},
+            "related_issue": self.issue_url,
         }
 
-    def fetch_issue_info(self, issue):
+    def is_open(self):
+        """Check if puul request is open."""
+        return self.state == "open"
+
+    def can_close(self):
+        """Check if pull request can be closed."""
+        return (
+            datetime.utcnow().replace(tzinfo=pytz.utc) - self.updated_at
+        ).days >= 3 * 30
+
+    def needs_comment(self):
+        """Check if pull request needs comment."""
+        comments = [comment for comment in self.comments()]
+        return not comments or comments[-1].user.login not in self.maintainers
+
+    def can_merge(self):
+        """Check if pull request can be merged."""
+        sha = self.statuses_url.rsplit("/", 1)[-1]
+        statuses = list(self.repository.statuses(sha))
+        return "error" not in [s.state for s in statuses]
+
+    def needs_review(self):
+        """Check if pull request needs review."""
+        reviews = [r for r in self.reviews()]
+        return len(reviews) == 0
+
+    filters = [is_open, can_close, needs_comment, can_merge, needs_review]
+
+    @property
+    def status(self):
+        """The status of the pull request."""
+        status = {}
+        for f in self.filters:
+            status[f.__name__] = f(self)
+        return status
+
+    @property
+    def actions(self):
+        """The suitable actions for the pull request."""
+        actions = []
+        if not self.status["is_open"]:
+            return actions
+        if self.status["needs_review"]:
+            actions.append("Review this!")
+        elif self.status["can_merge"]:
+            actions.append("Merge this!")
+        elif self.status["can_close"]:
+            actions.append("Close this!")
+        elif self.statusp["needs_comment"]:
```

Comment:
```suggestion
        elif self.statusp["needs_comment"]:
```
```suggestion
        elif self.status["needs_comment"]:
```

Check other places for this typo as well